pkg/pipeline/pipeline.go

```diff
@@ -149,31 +156,81 @@ func NewPipeline() (*Pipeline, error) {
 		Writer:       writer,
 	}
 
+	// TODO: all this apply* functions can be removed if we change the Ingester, Transformer, etc...
+	// interfaces and use them directly here
```

Comment:
@eranra @mariomac Let's discuss this in our next meeting. We need to decide on the order of merging PRs to make this transition.